app/scripts/directives/oscPersistentVolumeClaim.js

```diff
@@ -26,7 +27,26 @@ angular.module("openshiftConsole")
         }];
         scope.claim.selectedLabels = [];
         DataService.list({group: 'storage.k8s.io', resource: 'storageclasses'}, {}, function(storageClasses) {
-          scope.storageClasses = storageClasses.by('metadata.name');
+          scope.storageClasses = _.map(storageClasses.by('metadata.name'));
+          var defaultClassAnnotation = "storageclass.beta.kubernetes.io/is-default-class";
+          _.each(scope.storageClasses, function(sclass){
+            if ("annotations" in sclass.metadata && defaultClassAnnotation in sclass.metadata.annotations && sclass.metadata.annotations[defaultClassAnnotation] === "true") {
+              scope.defaultStorageClass = sclass.metadata.name;
+            }
+          });
+          if (scope.defaultStorageClass === "")  { //if there is no default, set a no storage class option
```

Comment:
I think we can leave the "no storage class" option even if there is a default as long as we say the default will be used in the help text. cc @jwforres opinion?